ludwig/experiment.py

```diff
@@ -496,6 +498,15 @@ def kfold_cross_validate(
                 raw_kfold_stats[category] = {}
             category_stats = \
                 kfold_cv_stats[fold_name]['fold_metric'][category]
+
+            # clear out predictions Tensors
+            # todo revisit to see if we can save predictions output
+            #    not enough structures to use output feature.postprocess_results()
+            keys = [key for key in category_stats
+                    if isinstance(category_stats[key], tf.Tensor)]
+            for key in keys:
+                del category_stats[key]
```

Comment:
Wouldn't

```
category_stats[key] = category_stats[key].numpy()
```

work?

Reply:
I initially thought of doing that as well.

For numerical feature, this would be workable.  

Categorical or sequence features, the above will return the encoded value for the `predictions` key, e.g., {'predictions'; [1, 1, 3, 2]}`.  OTOH `postprocess_results()` returns `'predictions': ['a', 'a', 'c', 'b']`.

On reflection, this might be a better work-around because this provides predictions in some form as opposed to no predictions plus  the `probabilities` array will be provided.  If we do this, I just need to make a note in the documentation, that `predictions` in the `fold_metric` dictionary is the encoded representation of the output feature.

This issue probably shows the need to add a few more tests to make sure other output feature types do not present other problems.

I'll proceed along this approach.

Reply:
My boubt is: the values in `fold_metric` are basically the values from the test stats for the different folds. Because of that, they should have already been postprocessed at this point, meaning all output tensors alrady converted etc. so this should not be an issue at this point, maybe we should go back in the steps to figure out why they are not postprocessed.

Reply:
I understand the point.  To answer your specific question
> maybe we should go back in the steps to figure out why they are not postprocessed.

The reason kfold_cv does not see post-processed values is that kfold_cv uses the `experiment()` method to generate predictions.  This method does not include a call to `postprocess_results()`. 
https://github.com/uber/ludwig/blob/ff7159992eb59d5e8cb2bc3b4850d8a10a91f702/ludwig/experiment.py#L135-L161

OTOH, the `full_experiment()` method includes both the call to `experiment()` and `postprocess_results()`.  
https://github.com/uber/ludwig/blob/ff7159992eb59d5e8cb2bc3b4850d8a10a91f702/ludwig/experiment.py#L314-L381.

When I initially looked to run `postprocess_results()` after `experiment()` in the kfold_cv function, it appeared that I did not have addressability to the `metadata` parameter.  I'll take another look to see if it is possible to run the `postprocess_result()` method.